src/main/java/io/github/moulberry/notenoughupdates/profileviewer/SacksPage.kt

```diff
@@ -0,0 +1,505 @@
+            for (item in contents) {
+                if (!item.isJsonPrimitive || !item.asJsonPrimitive.isString) return
+                val sackItem = item.asString
+
+                val adjustedName = sackItem.replace("-", ":")
+                val itemCount = sacksInfo.getInt(adjustedName)
+                val itemValue = itemCount * getPrice(sackItem)
+
+                if (sackItem !in sackItems) {
+                    totalValue += itemValue
+                    totalItems += itemCount
+                }
+
+                sackItems[sackItem] = SackItem(itemCount, itemValue)
+                sackValue += itemValue
+                sackItemCount += itemCount
+            }
+            sackContents[sackName] = SackInfo(sackItemCount, sackValue)
+        }
+
+        for ((itemName, _) in sacksInfo.entrySet()) {
+            val adjustedName = itemName.replace(":", "-")
+            if (adjustedName.contains(Regex("(RUNE|PERFECT_|MUSHROOM_COLLECTION)"))) continue
+            if (adjustedName in sackItems) continue
+            println("$adjustedName missing from repo sacks file!")
+        }
+
+        sackContents["All"] = SackInfo(totalItems, totalValue)
+    }
+
+    private fun getPrice(itemName: String): Double {
+        val npcPrice = HypixelItemAPI.getNPCSellPrice(itemName) ?: 0.0
+        val bazaarInfo = manager.auctionManager.getBazaarInfo(itemName) ?: return npcPrice
+        val buyPrice = bazaarInfo.getDouble("curr_buy")
+        val sellPrice = bazaarInfo.getDouble("curr_sell")
+        return maxOf(npcPrice, buyPrice, sellPrice)
+    }
+
+    private fun getRuneData(sacksInfo: JsonObject): Int {
+        var sackItemCount = 0
+        for ((itemName, amount) in sacksInfo.entrySet()) {
+            if (!amount.isJsonPrimitive || !amount.asJsonPrimitive.isNumber) continue
+            sackPattern.matchMatcher(itemName) {
+                val itemAmount = amount.asInt
+                val name = group("name")
+                val tier = group("tier")
+                val neuInternalName = "${name}_RUNE;$tier"
+                sackItemCount += itemAmount
+                sackItems[neuInternalName] = SackItem(itemAmount, 0.0)
+                playerRunes.add(neuInternalName)
+            }
+        }
+        sackContents["Rune"] = SackInfo(sackItemCount, 0.0)
+        return sackItemCount
+    }
+
+    private fun JsonObject.getInt(key: String): Int {
+        return if (has(key) && get(key).isJsonPrimitive && get(key).asJsonPrimitive.isNumber) {
+            get(key).asInt
+        } else {
+            0
+        }
+    }
+
+    private fun JsonObject.getDouble(key: String): Double {
+        return if (has(key) && get(key).isJsonPrimitive && get(key).asJsonPrimitive.isNumber) {
+            get(key).asDouble
+        } else {
+            0.0
+        }
+    }
+
+    private inline fun <T> Pattern.matchMatcher(text: String, consumer: Matcher.() -> T) =
+        matcher(text).let { if (it.matches()) consumer(it) else null }
```

Comment:
Those look like some handy utils, you might want to move them to a KotlinJsonUtil file of sorts. Additionally I would rename the getNumber methods to indicate that they return 0 as a fallback: getDoubleOrZero